src/app/components/homepage/homeform/homeform.component.ts

```diff
@@ -1,8 +1,44 @@
 import { Component } from '@angular/core';
+import { FormControl, FormGroup, Validators, AbstractControl, ValidatorFn } from '@angular/forms';
 
 @Component({
   selector: 'app-homeform',
   templateUrl: './homeform.component.html',
   styleUrls: ['./homeform.component.scss'],
 })
-export class HomeformComponent {}
+export class HomeformComponent {
+  loginForm = new FormGroup({
+    email: new FormControl('', [Validators.required, Validators.email]),
+    password: new FormControl('', Validators.required),
+  });
+
+  registerForm = new FormGroup({
+    firstname: new FormControl('', Validators.required),
+    lastname: new FormControl('', Validators.required),
+    email: new FormControl('', [Validators.required, Validators.email]),
+    password: new FormControl('', [Validators.required, this.passwordStrengthValidator()]),
```

Comment:
Optional, add a password validation by repeating the filed and controlling equalities between both values